src/components/typography/Text.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { CSSProperties, forwardRef, PropsWithChildren } from 'react';
+import { isDefined } from '~common/helpers/types';
+import { DesignTokensColorsText } from '~types/design-tokens';
+
+export enum TextSize {
+  Small = 'small',
+  Default = 'default',
+  Large = 'large',
+}
+
+interface Props {
+  as?: 'span' | 'p' | 'div';
+  className?: string;
+  colorOverride?: DesignTokensColorsText;
+  isStrong?: boolean;
+  isSubdued?: boolean;
+  size?: TextSize;
+}
+
+export const Text = forwardRef<HTMLSpanElement, PropsWithChildren<Props>>((props, ref) => {
+  const {
+    children,
+    colorOverride,
+    isStrong = false,
+    isSubdued = false,
+    size = TextSize.Default,
+    ...restAndRadixProps
+  } = props;
+
+  const additionalProps: { style?: CSSProperties } = isDefined(colorOverride)
+    ? { style: { color: `var(--${colorOverride})` } }
```

Comment:
Should we compose it with `restAndRadixProps` in case a style prop is passed? Or just invert the order in which we apply `restAndRadixProps` and `additionalProps` in the component?

Reply:
I'm not sure how to do this with our typing... It would be a bit hacky. Happy to discuss it

Reply:
mmmh yeah true, the typing is gonna be a pain for this, maybe let's just revert the restAndRadixProps and additionalProps, so that a custom style property would override it.